lang/src/main/java/org/apache/shiro/io/XmlSerializer.java

```diff
@@ -34,7 +34,9 @@
  * JavaBeans compatible!
  * 
  * @since 0.9
+ * @deprecated This class should not be used because of unsecure XMLEncoder/XMLDecoder usage.
```

Comment:
```suggestion
 * @deprecated This class should not be used directly because of unsecure XMLEncoder/XMLDecoder usage.
```

You might want to say "used directly" 🤔 
if/when this is wrapped in encryption most of the risk is removed.

Reply:
@bdemers good catch ;)